create .NET Interactive extension package

colombod opened the pull request:
Add .NET Interactive extension pakcage

cartermp commented on XPlot.Plotly.NET.Interactive/KernelExtension.fs:
```diff
@@ -0,0 +1,50 @@
+};"
+        ) |> ignore
+        newScript.AppendLine(JavascriptUtilities.GetCodeForEnsureRequireJs(Uri("https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js"), "renderPlotly")) |> ignore
+        newScript.ToString()
```
Just for fun, this can now be an interpolated string since it's using the .NET 5 SDK

cartermp reviewed (changes requested):
To get CI to pass I think we'll also have to change the tool manifest file and pull in the very latest version of paket and fake-cli

daxian-dbw commented on NuGet.config:
```diff
@@ -0,0 +1,17 @@
+    <add key="dotnet3-dev" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet3.1/nuget/v3/index.json" />
+    <add key="dotnet5" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet5/nuget/v3/index.json" />
+    <add key="MachineLearning" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/MachineLearning/nuget/v3/index.json" />
+    <add key="PSGallery" value="https://www.powershellgallery.com/api/v2/" />
```
Unless you are going to pull PowerShell modules in this repo, this feed probably should be removed.

colombod replied:
I think I have to implement in this repo the code that the powershell kernel was doing to integrate the xplot types. What do you think?

daxian-dbw replied:
Yes, the `New-PlotyChart` cmdlet needs to be moved here. But the `PSGallery` is not needed for that. You will need to reference to [`PowerShellStandard.Library`](https://www.nuget.org/packages/PowerShellStandard.Library/5.1.0).

daxian-dbw replied:
This entry should be removed.

colombod committed:
start


add extension method


update tools


updated build


update Fsharp.Formatting.CommandTool


update test project


update tfm


installed updates


move to right folder


add powershell kernel logic


wip

colombod committed:
xplot kernel extension targets XPlot.Plotly 3.0.1

colombod committed:
loading accelerator for powershell kernel support

colombod committed:
ps module

colombod committed:
complete commandlet

colombod committed:
explicit type

colombod committed:
renamed extension package

colombod committed:
Brett did things here and I got better

daxian-dbw commented on src/XPlot.Plotly.Interactive/NewPlotlyChartCommand .fs:
```diff
@@ -0,0 +1,31 @@
+
+    override _.BeginProcessing() = ()
+
+    override _.ProcessRecord() = ()
```
Please do the same as the original implementation. This makes sure the piping behavior is as expected.
```
        private List<Trace> _traces;

        /// <summary>
        /// BeginProcessing override.
        /// </summary>
        protected override void BeginProcessing()
        {
            _traces = new List<Trace>();
        }
        
        /// <summary>
        /// ProcessRecord override.
        /// </summary>
        protected override void ProcessRecord()
        {
            _traces.AddRange(Trace);
        }
```

colombod committed:
feedback from PR

daxian-dbw commented on src/XPlot.Plotly.Interactive/NewPlotlyChartCommand .fs:
```diff
@@ -18,13 +22,16 @@ type NewPlotlyChartCommand() =
+        
 
     override this.EndProcessing() =
         let traces = if isNull this.Trace then [||] else this.Trace
```
This line can be removed now

colombod committed:
feedback

colombod committed:
package version
